### ISSUE TITLE
run ONLY selected code in query tool

### ISSUE THREAD

**Issue by ccurvey** [commandprompt/pgmanage#229]
I thought this used to be here...can't find a setting....

Let's say I'm working in the query tool, and I type 

```
create table foo (a int)
```

I run the query, and it works.  Great.  Now I add more code to the query tool

```
create table foo (a int)

select * 
from foo
```

Now I highlight `select * from foo` and run the query (1).  And I get an error message:  

`syntax error at or near "select" LINE 3: select * from foo ^`

Well, that's odd....let me go throw a semicolon in there

```
create table foo (a int);

select * 
from foo
```

and I get another error message saying

`relation "foo" already exists`

What that tells me is that the Query Tool is sending the ENTIRE stream of commands, not just the one I highlighted.  Color me surprised!

I'd really like to see this implemented for two reasons

Reason 1:  My example shows a non-destructive case.  But imagine if I wrote

```
-- increase my prices
update product set price = price * 1.1;

-- check out my results
select * from product
```

ohhh, that would be bad!  I just raised my prices by 21%, and not 10% -- **and it was done silently**

Reason 2:  it is often useful to be able to see the results of one query, while formulating another.  for example

```
select * from purchase   -- which shows a customer id

select * from customer where id in (123, 456, 789)
```

(1) I remap the "query execute" keystroke to ctrl-enter, if that makes any difference....

**Comment by commandprompt** [comment]
Hi @ccurvey.
This seems to be a good idea for enhancement, perhaps we could have two buttons **Run** AND **Run Selection**, I need to think on that.

**Comment by ccurvey** [comment]
Please do.  It really makes a difference.  Don't forget to have a keyboard
shortcut for "run selected". :)

**Comment by eug3nix** [comment]
implemented in https://github.com/commandprompt/pgmanage/commit/5744fd37e8221eb899f3c2e171d37d5be4abc5da

**Comment by eug3nix** [comment]
Hi.
This feature was implemented in 1.0rc1.
The "run selection" can be called in 3 ways:
1. by using a new button in the query tab toolbar
2. using a hotkey
3. by right-clicking in the sql code editor and selecting the corresponding context menu item.
Note: if there is no selected region, the whole line under the cursor will be executed.

Thank you for your feedback!

**Comment by ccurvey** [comment]
Thank you so much.  Someday when you're bored, it would be great if the app could detect if there is a selection made, and if so, ONLY run the selected text.
